standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetastoreTServerSocket.java

```diff
@@ -29,8 +32,32 @@
  * accepted sockets.
  *
  */
-public class TServerSocketKeepAlive extends TServerSocket {
-  public TServerSocketKeepAlive(TServerSocket serverSocket) throws TTransportException {
+public class MetastoreTServerSocket extends TServerSocket {
+  private final Configuration configuration;
+
+  public MetastoreTServerSocket(TServerSocket serverSocket,
+      Configuration conf) throws TTransportException {
     super(serverSocket.getServerSocket());
+    this.configuration = Objects.requireNonNull(conf);
+  }
+
+  @Override
+  public TSocket accept() throws TTransportException {
+    TSocket ts = super.accept();
+    boolean keepAlive = MetastoreConf.getBoolVar(configuration, MetastoreConf.ConfVars.TCP_KEEP_ALIVE);
+    if (keepAlive) {
+      try {
+        ts.getSocket().setKeepAlive(true);
+      } catch (SocketException e) {
+        throw new TTransportException(e);
+      }
+    }
+    // get the limit from the configuration for every new connection
+    int maxThriftMessageSize = (int) MetastoreConf.getSizeVar(
+        configuration, MetastoreConf.ConfVars.THRIFT_METASTORE_CLIENT_MAX_MESSAGE_SIZE);
+    if (maxThriftMessageSize > 0) {
+      ts.getConfiguration().setMaxMessageSize(maxThriftMessageSize);
```

Comment:
should we check if `ts.getConfiguation` != null similar to `HiveAuthUtils.configureThriftMaxMessageSize`

Reply:
the configuration in TSocket is not null for all cases